editor/src/app/sites/sites-state/sites.state.ts

```diff
@@ -37,94 +45,120 @@ export class SitesState implements NgxsOnInit {
 
 
   @Action(CreateSiteAction)
-  createSite({ setState, getState }: StateContext<SiteStateModel[]>) {
-    const currentState = getState();
-    const newSite: SiteStateModel = {
-      // @todo sync with backend
-      // @todo get unique name from backend
-      name: 'untitled-' + Math.random().toString(36).substr(2, 9),
-      title: '',
-      order: currentState.length,
-      '@attributes': {
-        published: 0
-      }
-    };
-
-    setState(
-      [...currentState, newSite]
-    );
+  createSite({ setState, getState }: StateContext<SiteStateModel[]>, action: CreateSiteAction) {
+    const siteName = action.site ? (action.site.name === '' ? '0' : action.site.name) : '-1';
+    this.appStateService.sync('sites', { site: siteName }, 'POST')
+      .then((response: any) => {
+        if (response.error_message) {
+          // @TODO handle error message
+          console.error(response.error_message);
+        } else {
+          const currentState = getState();
+          const newSite: SiteStateModel = response.site;
+
+          setState(
+            [...currentState, newSite]
+          );
+
+          if (response.settings) {
+            this.store.dispatch(new CreateSiteSettingsAction(newSite, response.settings));
+          }
+
+          if (response.siteTemplateSettings) {
+            this.store.dispatch(new CreateSiteTemplateSettingsAction(newSite, response.siteTemplateSettings));
+          }
+
+          if (response.sections && response.sections.length) {
+            this.store.dispatch(new AddSiteSectionsAction(response.sections));
+          }
+
+          if (response.entries && response.entries.length) {
+            this.store.dispatch(new AddSiteEntriesAction(newSite, response.entries));
+          }
+
+          if (response.tags && response.tags.section) {
+            this.store.dispatch(new AddSiteSectionsTagsAction(newSite, response.tags));
+          }
+        }
+      });
   }
 
 
   @Action(UpdateSiteAction)
   updateSite({ setState, getState }: StateContext<SiteStateModel[]>, action: UpdateSiteAction) {
     const currentState = getState();
+    const path = 'site/' + action.site.order + '/' + action.field.split('.').join('/');
+    const data = {
+      path: path,
+      value: action.value
+    };
 
-    // @todo sync with backend, set returned `name` from server (already slugified and unique)
-    setState(
-      currentState.map((site) => {
-        if (site.name === action.site.name) {
-          return set(action.field, action.value, site);
+    this.appStateService.sync('sites', data)
+      .then((response: any) => {
```

Comment:
Let's keep to observables since we're already using them everywhere. So return observable from `sync` (remove the `toPromise`). And add `.pipe(map(response => {...}))` here.